File: 728-self-dividing-numbers/self-dividing-numbers.cpp
class Solution {
public:
    vector<int> selfDividingNumbers(int l, int r) {
        vector<int> res;
        int dig;int temp;int flag;
        for(int i=l;i<=r;i++){
            temp=i;
            flag=1;
            while (temp!=0){
                dig=temp%10;
                if(dig==0 || i%dig!=0){
                    flag=0;break;
                }
                temp=temp/10;
            }
            if (flag){
                res.push_back(i);
            }
        }
        return res;
    }
};

File: 728-self-dividing-numbers/self-dividing-numbers.py
class Solution(object):
    def selfDividingNumbers(self, left, right):
        """
        :type left: int
        :type right: int
        :rtype: List[int]
        """
        res=[]
        flag=1
        for i in range(left,right+1):
            st = str(i)
            flag = 1
            for j in st:
                if (int(j)==0 or i%int(j)!=0):
                    flag = 0
                    break
            if(flag):
                res.append(i)
        return res


        